ghost/core/core/server/services/posts/stats/post-stats.js

```diff
@@ -0,0 +1,31 @@
+class PostStats {
+    /**
+     * @param {Object} config
+     * @param {Object} config.db - an instance holding knex connection to the database
+     * @param {Boolean} config.isSQLite - flag identifying if storage is connected to SQLite
+     */
+    constructor({db, isSQLite}) {
+        this._db = db;
+        this._isSQLite = isSQLite;
+    }
+
+    /**
+     * Returns the most recently `published_at` post that was published or sent
+     * via email
+     */
+    async getMostRecentlyPublishedPostDate() {
+        const result = await this._db.knex.raw('SELECT * FROM posts WHERE status="published" OR status="sent" ORDER BY published_at DESC LIMIT 1');
```

Comment:
If you use knex `select|where|limit` methods rather than `raw`, I think you won't have to check for the database type as knex will handle under the hood, something like `knex('posts').where('blah', ....`